File: src/main/java/org/mitallast/queue/common/component/ModulesBuilder.java
package org.mitallast.queue.common.component;

import com.google.inject.Guice;
import com.google.inject.Injector;
import com.google.inject.Module;

import java.util.ArrayList;
import java.util.List;

public class ModulesBuilder {
    List<Module> modules = new ArrayList<>();

    public ModulesBuilder add(Module... modules) {
        for (Module module : modules) {
            add(module);
        }
        return this;
    }

    public ModulesBuilder add(Module module) {
        modules.add(module);
        return this;
    }

    public Injector createInjector() {
        return Guice.createInjector(modules);
    }

    public Injector createChildInjector(Injector injector) {
        return injector.createChildInjector(modules);
    }
}

File: src/main/java/org/mitallast/queue/common/component/ModulesBuilder.kt
package org.mitallast.queue.common.component

import com.google.inject.Guice
import com.google.inject.Injector
import com.google.inject.Module
import java.util.*

class ModulesBuilder {
    private var modules: MutableList<Module> = ArrayList()

    fun add(vararg modules: Module): ModulesBuilder {
        for (module in modules) {
            add(module)
        }
        return this
    }

    fun add(module: Module): ModulesBuilder {
        modules.add(module)
        return this
    }

    fun createInjector(): Injector {
        return Guice.createInjector(modules)
    }

    fun createChildInjector(injector: Injector): Injector {
        return injector.createChildInjector(modules)
    }
}
